app/recipes/routes.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python3
+import html
+import math
+import re
+from app import mongo
+from bson.objectid import ObjectId
+from datetime import datetime
+from flask import Blueprint, current_app, render_template, redirect, request, url_for, flash, session, Markup
+from slugify import slugify
+
+
+# --------------------- #
+#    Flask Blueprint    #
+# --------------------- #
+recipes = Blueprint("recipes", __name__)
+
+
+# -------------------- #
+#    DB Collections    #
+# -------------------- #
+allergens_collection = mongo.db.allergens
+desserts_collection = mongo.db.desserts
+measurements_collection = mongo.db.measurements
+recipes_collection = mongo.db.recipes
+users_collection = mongo.db.users
+
+
+# ------------------- #
+#    Global Helper    #
+# ------------------- #
+@recipes.context_processor
+def desserts_total():
+    desserts_count = recipes_collection.count
+    return dict(desserts_count=desserts_count)
+
+
+# ---------------------- #
+#    Helper Functions    #
+# ---------------------- #
+
+# Allergens Dropdown List
+def dropdown_allergens():
+    return sorted([item for allergen in allergens_collection.find() for item in allergen.get("allergen_name")])
+
+
+# Dessert Type Dropdown List
+def dropdown_dessert_type():
+    return sorted([item for dessert in desserts_collection.find() for item in dessert.get("dessert_type")])
+
+
+# Measurements Dropdown List
+def dropdown_measurement():
+    return [item for measurement in measurements_collection.find() for item in measurement.get("measurement_unit")]
+
+
+# Find Recipe by ObjectID
+def get_recipe_id(recipe_id):
+    return recipes_collection.find_one({"_id": ObjectId(recipe_id)})
+
+
+# Find user's lowercase username
+def get_user_lower(user_param):
+    return users_collection.find_one({"username_lower": user_param.lower()})
+
+
+# ---------------- #
+#    APP ROUTES    #
+# ---------------- #
+
+# ------------------------------------------- #
+#    CRUD: Create | Read | Update | Delete    #
+# ------------------------------------------- #
+
+# ----- CREATE ----- #
+@recipes.route("/desserts/new", methods=["GET", "POST"])
+def desserts_new():
+    """
+    CRUD - Create recipe for database.
+
+    Inject all form data to new recipe document on submit.
+    """
+    if request.method == "GET":
+        allergen_list = dropdown_allergens()
+        dessert_list = dropdown_dessert_type()
+        measurement_list = dropdown_measurement()
+        return render_template(
+            "desserts_new.html",
+            allergens=allergen_list,
+            desserts=dessert_list,
+            measurements=measurement_list)
+
+    if request.method == "POST":
+        # get today's date and date recipe was last edited
+        today = datetime.now().strftime("%d %B, %Y")
+        last_edit = int(datetime.now().strftime("%Y%m%d"))

```

Comment:
Why save `today` as string and `last_edit` as int? You should save them both as datetime objects and do any strftime conversion when you want to display the recipe. In that case you can do datetime operations more easily.